File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        maj, freq = 0, 0

        for num in nums:
            if freq == 0:
                maj = num
            if num == maj:
                freq += 1
            else:
                freq -= 1
            # print(maj, freq)

        return maj
        

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int maj = 0;
        int freq = 0;
        
        for (int num: nums) {
            if (freq == 0) {
                maj = num;
            }
            if (maj == num) {
                freq++;
            } else {
                freq--;
            }
        }
        return maj;
    }
}